src/test/java/codeu/controller/ActivityFeedServletTest.java

```diff
@@ -1,34 +1,185 @@
+    conversationList.add(CONVERSATION_ONE);
+    mockConversationStore.setConversations(conversationList);
+    activityFeedServlet.setConversationStore(mockConversationStore);
+    activityFeedServlet.setMessageStore(mockMessageStore);
+    activityFeedServlet.setUserStore(mockUserStore);
+
+    List<Activity> allActivities = new LinkedList<>();
+    String messageOneContent = "test_username_one sent in conversation_one: message one";
+    String creationTimeOne = "1969-12-31 18:00:01";
+    String creationTimeTwo = "1969-12-31 18:00:02";
+    String creationTimeThree = "1969-12-31 18:00:03";
+    allActivities.add(new Activity(creationTimeOne, messageOneContent));
+    String userOneContent = "test_username_one joined Charmer Chat.";
+    allActivities.add(new Activity(creationTimeTwo, userOneContent));
+    String conversationOneContent = "test_username_one started conversation conversation_one";
+    allActivities.add(new Activity(creationTimeThree, conversationOneContent));
+
+    activityFeedServlet.doGet(mockRequest, mockResponse);
+    Mockito.verify(mockRequest).setAttribute("activities", allActivities);
+  }
+
+  @Test
+  public void testDoGetUserOnly() throws IOException, ServletException {
+    final List<User> userList = new ArrayList<>();
+    userList.add(USER_ONE);
+    mockUserStore.setUsers(userList);
+    activityFeedServlet.setUserStore(mockUserStore);
+
+    List<Activity> allActivities = new LinkedList<>();
+    String creationTimeTwo = "1969-12-31 18:00:02";
+    String userOneContent = "test_username_one joined Charmer Chat.";
+    allActivities.add(new Activity(creationTimeTwo, userOneContent));
+
+    activityFeedServlet.doGet(mockRequest, mockResponse);
+    Mockito.verify(mockRequest).setAttribute("activities", allActivities);
+  }
+
+  @Test
+  public void testDoGetUserAndConversationOnly() throws IOException, ServletException {
+    final List<User> userList = new ArrayList<>();
+    userList.add(USER_ONE);
+    mockUserStore.setUsers(userList);
+    final List<Conversation> conversationList = new ArrayList<>();
+    conversationList.add(CONVERSATION_ONE);
+    mockConversationStore.setConversations(conversationList);
+    activityFeedServlet.setConversationStore(mockConversationStore);
+    activityFeedServlet.setUserStore(mockUserStore);
+
+    List<Activity> allActivities = new LinkedList<>();
+    String creationTimeTwo = "1969-12-31 18:00:02";
+    String creationTimeThree = "1969-12-31 18:00:03";
+    String userOneContent = "test_username_one joined Charmer Chat.";
+    allActivities.add(new Activity(creationTimeTwo, userOneContent));
+    String conversationOneContent = "test_username_one started conversation conversation_one";
+    allActivities.add(new Activity(creationTimeThree, conversationOneContent));
+
+    activityFeedServlet.doGet(mockRequest, mockResponse);
+    Mockito.verify(mockRequest).setAttribute("activities", allActivities);
+  }
+
+  /*@After
```

Comment:
This probably shouldn't be commented out. Why is it commented out? I would expect you need it